src/main/java/baritone/pathing/movement/MovementHelper.java

```diff
@@ -49,6 +49,9 @@
 public interface MovementHelper extends ActionCosts, Helper {
 
     static boolean avoidBreaking(BlockStateInterface bsi, int x, int y, int z, IBlockState state) {
+        if (!bsi.worldBorder.canPlaceAt(x, y)) {
+            return false;
+        }
```

Comment:
Shouldn't this return true to not allow breaking blocks outside the world border?

Reply:
Yea sorry, I don't quite know how I missed this